File: kotlin/sprint5_final/B/Solution.kt
/**
 * Comment it before submitting
 *
 * class Node(var left: Node?, var right: Node?, var value: Int)
 */

fun remove(root: Node?, key: Int): Node? {
    // Your code
    // ヽ(´▽`)/
}

fun test() {
    val node1 = Node(null, null, 2)
    val node2 = Node(node1, null, 3)
    val node3 = Node(null, node2, 1)
    val node4 = Node(null, null, 6)
    val node5 = Node(node4, null, 8)
    val node6 = Node(node5, null, 10)
    val node7 = Node(node3, node6, 5)
    val newHead = remove(node7, 10)
    assert(newHead!!.value == 5)
    assert(newHead.right == node5)
    assert(newHead.right!!.value == 8)
}

File: swift/sprint5_final/B/solution.swift
/*
 Comment it before submitting

 open class Node {
     
     public var value: Int
     public var left: Node?
     public var right: Node?
     
     public init(_ value: Int) {
         self.value = value
         self.right = nil
         self.left = nil
     }
 }
 */


func remove(root: Node?, key: Int) -> Node? {
    // Your code
    // ヽ(´▽`)/
    return root
}

func test() {
    var node1 = Node(2)
    var node2 = Node(3)
    node2.left = node1
    var node3 = Node(1)
    node3.right = node2
    var node4 = Node(6)
    var node5 = Node(8)
    node5.left = node4
    var node6 = Node(10)
    node6.left = node5
    var node7 = Node(5)
    node7.left = node3
    node7.right = node6
    var newHead = remove(root: node7, key: 10)
    assert(newHead!.value == 5)
    assert(newHead!.right === node5)
    assert(newHead!.right!.value == 8)
}
